test/AbstractWriter.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+import {assert} from 'chai';
+import * as ion from '../src/IonTests';
+
+function testWriteValue(reader, expected) {
+    let writer = ion.makeTextWriter();
+    writer.writeValue(reader);
+    assert.equal(String.fromCharCode.apply(null, writer.getBytes()), expected);
+}
+
+function testWriteValues(reader, expected) {
+    let writer = ion.makeTextWriter();
+    writer.writeValues(reader);
+    assert.equal(String.fromCharCode.apply(null, writer.getBytes()), expected);
+}
+
+function depthTest (instructions, expectedDepth) {
+        let textWriter = ion.makeTextWriter();
+        let binaryWriter = ion.makeBinaryWriter();
+        instructions(textWriter);
+        instructions(binaryWriter);
+        assert.equal(textWriter.depth(), expectedDepth);
+        assert.equal(binaryWriter.depth(), expectedDepth);
+}
+
+describe('Depth tests', () => {
```

Comment:
I'm not sure why this is polluting the diff... but this is the generalized test that you were asking for.
I'm going to leave the intern tests as is because we will be removing them entirely this week.